core/math/random_pcg.h

```diff
@@ -45,7 +45,10 @@ class RandomPCG {
 
 	RandomPCG(uint64_t seed = DEFAULT_SEED, uint64_t inc = PCG_DEFAULT_INC_64);
 
-	_FORCE_INLINE_ void seed(uint64_t seed) { pcg.state = seed; }
+	_FORCE_INLINE_ void seed(uint64_t seed) {
+		pcg.state = seed;
+		pcg32_random_r(&pcg);
```

Comment:
Can you add a comment explaining why we have to do this when seeding?

Reply:
Done